specification/ai/OpenAI.Assistants/run_steps/models.tsp

```diff
@@ -165,3 +165,181 @@ model RunStepError {
   @doc("The human-readable text associated with this error.")
   message: string;
 }
+
+//
+// The following types facilitate streaming use.
+//
+
+/** Represents a run step delta i.e. any changed fields on a run step during streaming. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+model RunStepDeltaChunk {
+  /** The identifier of the run step, which can be referenced in API endpoints. */
+  id: string;
+
+  /** The object type, which is always `thread.run.step.delta`. */
+  object: "thread.run.step.delta";
+
+  /** The delta containing the fields that have changed on the run step. */
+  delta: RunStepDeltaChunkObject;
+}
+
+/** Represents the delta payload in a streaming run step delta chunk. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+model RunStepDeltaChunkObject {
+  /** The details of the run step. */
+  @encodedName("application/json", "step_details")
+  stepDetails?: RunStepDeltaDetail[];
+}
+
+/** Represents a single run step detail item in a streaming run step's delta payload. */
+@discriminator("type")
+@added(ServiceApiVersions.vFuturePlaceholder)
+model RunStepDeltaDetail {
+  /** The object type for the run step detail object. */
+  type: string;
+}
+
+/** Represents a message creation within a streaming run step delta. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+model RunStepDeltaMessageCreation extends RunStepDeltaDetail {
+  /** The object type, which is always "message_creation." */
+  type: "message_creation";
+
+  /** The message creation data. */
+  @encodedName("application/json", "message_creation")
+  messageCreation: RunStepDeltaMessageCreationObject;
+}
+
+/** Represents the data within a streaming run step message creation response object. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+model RunStepDeltaMessageCreationObject {
+  /** The ID of the newly-created message. */
+  @encodedName("application/json", "message_id")
+  messageId?: string;
+}
+
+/** Represents an invocation of tool calls as part of a streaming run step. */
+@added(ServiceApiVersions.vFuturePlaceholder)
+model RunStepDeltaToolCallObject extends RunStepDeltaDetail {
+  /** The object type, which is always "tool_calls." */
+  type: "tool_calls";
+
+  /** The collection of tool calls for the tool call detail item. */
+  @encodedName("application/json", "tool_calls")
+  toolCalls?: RunStepDeltaToolCall[];
+}
+
+/** The abstract base representation of a single tool call within a streaming run step's delta tool call details. */
+@discriminator("type")
+@added(ServiceApiVersions.vFuturePlaceholder)
+model RunStepDeltaToolCall {
+  /** The index of the tool call detail in the run step's tool_calls array. */
+  index: int32;
+
+  /** The ID of the tool call, used when submitting outputs to the run. */
+  id: string;
+
+  /** The type of the tool call detail item in a streaming run step's details. */
+  type: string;
+}
```

Comment:
Similar to a previous comment, I would be a bit wary of lifting too many fields to the super "class" (model) just so that we don't code ourselves into a corner. In this case, everything appears to be pretty standard, but still, `id` and `index` could be safer to repeat them in the models extending from this one.

Reply:
This one I'm comfortable with because they're functionally required in all cases -- `index` is necessary to place in the overall `tool_calls` array aggregation and `id` is needed to perform any client-service association. OAI likely omits any common intervening layer here since the commonality is so small, but since we need it anyway (for now) for the best class hierarchy emitter support, it's worthwhile to keep here.